src/ray/common/ray_syncer/node_state.cc

```diff
@@ -67,6 +73,9 @@ bool NodeState::ConsumeSyncMessage(std::shared_ptr<const RaySyncMessage> message
   }
 
   current = message;
+  if (on_raylet_rpc_completion_) {
+    on_raylet_rpc_completion_(NodeID::FromBinary(message->node_id()));
+  }
```

Comment:
I think the best place to trigger the callback is `RaySyncerBidiReactorBase::OnReadDone()` because `NodeState::ConsumeSyncMessage` can be called in different settings.

Reply:
Sounds good, I think it's good to update right after a successful grpc call.

Reply:
Ok I recall why I put it in the `NodeState` at the first place, reactors are stored per-node as a map inside of `RaySyncer`: https://github.com/ray-project/ray/blob/2e4a126deecf84d2930ad772e763e206e2f5f7d2/src/ray/common/ray_syncer/ray_syncer.h#L162-L166

which means if you tie the logic grpc callback, you have to store it somewhere and pass it to every reactor construction.